modules/account-lib/src/coin/stx/contractBuilder.ts

```diff
@@ -0,0 +1,131 @@
+    const txData = tx.toJson();
+    if (txData.payload === undefined) {
+      throw new InvalidTransactionError('payload must not be undefined');
+    }
+    if (txData.payload.payloadType == PayloadType.ContractCall) {
+      this.contractAddress(txData.payload.contractAddress)
+      this.contractName(txData.payload.contractName)
+      this.functionName(txData.payload.functionName)
+      this.functionArgs(txData.payload.functionArgs)
+      super.initBuilder(tx);
+    } else {
+      throw new BuildTransactionError("Transaction should be contract call")
+    }
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    this._options = this.buildContractCallOptions();
+    this.transaction.setTransactionType(TransactionType.ContractCall);
+    this.transaction.stxTransaction = await makeUnsignedContractCall(this._options);
+    return await super.buildImplementation();
+  }
+
+  private buildContractCallOptions(): UnsignedContractCallOptions | UnsignedMultiSigContractCallOptions {
+    const defaultOpts: ContractCallOptions = {
+      contractAddress: this._contractAddress,
+      contractName: this._contractName,
+      functionName: this._functionName,
+      functionArgs: this._functionArgs,
+      network: this._network,
+      fee: new BigNum(this._fee.fee),
+      nonce: new BigNum(this._nonce),
+    };
+    if (this._multiSignerKeyPairs.length === 0) {
+      throw new InvalidParameterValueError('supply at least 1 public key');
+    }
+    if (this._multiSignerKeyPairs.length === 1) {
+      return {
+        ...defaultOpts,
+        publicKey: this._multiSignerKeyPairs[0].getKeys(this._multiSignerKeyPairs[0].getCompressed()).pub,
+      };
+    } else {
+      return {
+        ...defaultOpts,
+        publicKeys: this._multiSignerKeyPairs.map(k => k.getKeys(k.getCompressed()).pub),
+        numSignatures: this._numberSignatures,
+      };
+    }
+  }
+
+  //region Contract fields
+  /**
+   * Set the contract address
+   *
+   * @param {string} address the address deployed the contract
+   * @returns {TransferBuilder} the builder with the new parameter set
+   */
+  contractAddress(address: string): this {
+    if (!isValidAddress(address)) {
+      throw new InvalidParameterValueError('Invalid address');
+    }
+    this._contractAddress = address;
+    return this;
+  }
+
+  /**
+   * Set the contract name
+   *
+   * @param {string} name name of contract
+   * @returns {TransferBuilder} the builder with the new parameter set
+   */
+  contractName(name: string): this {
+    if (name.length == 0) {
+      throw new InvalidParameterValueError('Invalid name');
+    }
+    this._contractName = name;
+    return this;
+  }
+
+  /**
+  * Set the function name in contract
+  *
+  * @param {string} name name of function
+  * @returns {TransferBuilder} the builder with the new parameter set
+  */
+  functionName(name: string): this {
+    if (name.length == 0) {
```

Comment:
Similarly @tylerlevine should we restrict the function names to `stack-stx`, `delegate-stx`, `delegate-stack-stx`, `stack-aggregation-commit`, and `revoke-delegate-stx` here as per https://bitgoinc.atlassian.net/browse/STLX-1898?

Reply:
Same as above - if that's all we need to support then restricting to those values is fine with me.